pennylane_qiskit/vqe/vqe_runner.py

```diff
@@ -0,0 +1,512 @@
+                "default": "EfficientSU2",
+            },
+            "ansatz_config": {
+                "description": "Configuration parameters for the ansatz circuit.",
+                "type": "dict",
+            },
+            "optimizer": {
+                "description": "Classical optimizer to use, default='SPSA'.",
+                "type": "string",
+                "default": "SPSA",
+            },
+            "optimizer_config": {
+                "description": "Configuration parameters for the optimizer.",
+                "type": "dict",
+            },
+            "shots": {
+                "description": "The number of shots used for each circuit evaluation.",
+                "type": "integer",
+            },
+            "use_measurement_mitigation": {
+                "description": "Use measurement mitigation, default=False.",
+                "type": "boolean",
+                "default": False,
+            },
+        },
+        "required": ["hamiltonian", "x0"],
+    }
+
+    meta["spec"]["return_values"] = {
+        "$schema": "https://json-schema.org/draft/2019-09/schema",
+        "description": "Final result in SciPy optimizer format",
+        "type": "object",
+    }
+
+    meta["spec"]["intermadiate_results"] = {
+        "$schema": "https://json-schema.org/draft/2019-09/schema",
+        "description": "Dictionnary containing: "
+        "The number of evaluation at current optimization step."
+        "Parameter vector at current optimization step."
+        "Function value at the current optimization step."
+        "The size of the step.",
+        "type": "dict",
+    }
+
+    provider = IBMQ.get_provider(hub=hub, group=group, project=project)
+
+    program_id = provider.runtime.upload_program(
+        data="pennylane_runtime/vqe_runtime.py", metadata=meta
+    )
+    return program_id
+
+
+def delete_vqe_runner(provider, program_id):
+    r"""Delete the desired program on IBMQ platform.
+    Args:
+        provider (object): IBMQ provider.
+        program_id (str): Id of the qiskit runtime to be deleted.
+    """
+    provider.runtime.delete_program(program_id)
+
+
+def vqe_runner(
+    backend,
+    hamiltonian,
+    x0,
+    program_id,
+    ansatz="EfficientSU2",
+    ansatz_config=None,
+    optimizer="SPSA",
+    optimizer_config=None,
+    shots=8192,
+    use_measurement_mitigation=False,
+    **kwargs,
+):
+    """Routine that executes a given VQE problem via the sample-vqe program on the target backend.
+
+    Parameters:
+        backend (ProgramBackend): Qiskit backend instance.
+        hamiltonian (qml.Hamiltonian): Hamiltonian whose ground state we want to find.
+        program_id(str): Optional, if the program is already uploaded on your account, you should provide
+            the corresponding id, otherwise a new program will be uploaded.
+        ansatz (Quantum function or str): Optional, a PennyLane quantum function or the name of the Qiskit
```

Comment:
Here it is specified that if the program id is not given, it will be created. However, this is not checked at any time.

Reply:
Mistake from old version!